.devcontainer/init.py

```diff
@@ -13,3 +14,10 @@
     REQUIREMENTS: list[str] = MANIFEST["requirements"]
 
 pip.main(PIP_ARGUMENTS + REQUIREMENTS)
+
+# Clone the Home Assistant core repo in the container homedir and symlink it in
+# custom_components, so that VSCode can follow the homeassistant.* Python imports.
+os.system("""
+    git clone --depth 1 https://github.com/home-assistant/core.git "${HOME}/homeassistant" &&
+    ln -s "${HOME}/homeassistant" custom_components
+""")
```

Comment:
These commands could alternatively go in a `Dockerfile` `RUN` instruction or separate shell script, or perhaps Python `subprocess.run()` and `os.symlink()`, but as above it also does the job.